R/report.R

```diff
@@ -187,7 +186,32 @@ pkg_source_link <- function(chk) {
   if (!is.null(chk$cran)) {
     paste0("https://github.com/cran/", chk$package)
   } else {
-    "???"
+    NA
+  }
+}
+
+pkg_github <- function(desc) {
+  urls <- c(
+    desc$get_field("BugReports", default = character()),
+    desc$get_urls()
+  )
+  gh_links <- grep("^https?://github.com/", urls, value = TRUE)
+
+  if (length(gh_links) == 0) {
+    NA_character_
+  } else {
+    re <- paste0(
+      "^",
+      "(?:https?://github.com/)",
+      "(?<owner>[^/]+)/",
+      "(?<repo>[^/#]+)",
+      "/?",
+      "(?<fragment>.*)",
+      "$"
+    )
+
+    remote <- rematch2::re_match(gh_links[[1]], re)
+    paste0("https://github.com/", remote$owner, "/", remote$repo)
```

Comment:
FWIW, remotes has some code for this: https://github.com/r-lib/remotes/blob/f7e61a44e764c52c96d4839d022b4b542bc4386c/R/install-dev.R#L40-L74

If you want to handle gitlab, etc.

Reply:
If there is no match, then this will return something like `https://github.com/NA/NA`, is that ok? 

We also don't handle multiple matches, but that is possibly fine.

Reply:
It's already split up into comma delimited pieces above, and there's a basic check. So this will only give `https://github.com/NA/NA` for weirdly formed github urls (which I think is fine, and I'll fix if I discover)

Reply:
ok